sql/core/src/main/scala/org/apache/spark/sql/execution/command/views.scala

```diff
@@ -207,29 +210,35 @@ case class CreateViewCommand(
   }
 
   /**
-   * Returns a [[CatalogTable]] that can be used to save in the catalog. This comment canonicalize
-   * SQL based on the analyzed plan, and also creates the proper schema for the view.
+   * Returns a [[CatalogTable]] that can be used to save in the catalog. Generate the view-specific
+   * properties(e.g. view default database, view query output column names) and store them as
+   * properties in the CatalogTable, and also creates the proper schema for the view.
+   *
+   * @param session the spark session.
+   * @param aliasedPlan if `userSpecifiedColumns` is defined, the aliased plan outputs the user
+   *                    specified columns, else it is the same as the `analyzedPlan`.
+   * @param analyzedPlan the analyzed logical plan that represents the child of a view.
```

Comment:
We generate the `queryColumnNames` by `analyzedPlan`, and we generate the view schema by `aliasedPlan`, they are not the same when `userSpecifiedColumns` is defined.